Mathlib/Algebra/Order/Monoid/Unbundled/Units.lean

```diff
@@ -0,0 +1,39 @@
+/-
+Copyright (c) 2025 Kenny Lau. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kenny Lau
+-/
+
+import Mathlib.Algebra.Group.Units.Defs
+import Mathlib.Algebra.Order.Monoid.Canonical.Defs
+
+
+/-!
+# Lemmas for units in an ordered monoid
+-/
+
+variable {M : Type*} [Monoid M] [LE M]
+
+theorem Units.mulLECancellable_val [MulLeftMono M]
+    (a : Mˣ) : MulLECancellable (↑a : M) := fun _ _ h ↦ by
+  simpa using mul_le_mul_left' h ↑a⁻¹
+
+theorem Units.mul_le_mul_left [MulLeftMono M]
+    (a : Mˣ) {b c : M} : a * b ≤ a * c ↔ b ≤ c :=
+  a.mulLECancellable_val.mul_le_mul_iff_left
```

Comment:
Given the name of this lemma, I think all of the new theorems in this file should be called `mul_le_mul_iff_left` instead of `mul_le_mul_left`

Reply:
Great, so much for consistency... I agree your name is reasonable.